doc/sphinx/how-to/how-to-test-mir-for-a-release.md

```diff
@@ -0,0 +1,174 @@
+# How to test Mir for a release
+The following document details the criteria that must be met before we can
+publish an official release of Mir. When adding to this test plan, please
+do not fail to consider how your additions may be automated in the future, as
+this is the end goal of our release testing.
+
+Note that all of these tests must be run _after_ a release branch is created
+in the Mir repository, such that the right repositories are updated beforehand.
+
+## Setup
+1. Install mir from the release candidate PPA
+    ```
+    sudo add-apt-repository --update ppa:mir-team/rc
+    sudo apt install mir-demos mir-platform-graphics-virtual mir-graphics-drivers-desktop mir-test-tools
+    ```
+2. Install the `mir-test-tools` from the `22/beta` _or_ `24/beta` track (depending
+   on which you are trying to test):
+    ```
+    sudo snap refresh
+    sudo snap install mir-test-tools --channel=24/beta
+    # ... or ...
+    sudo snap install mir-test-tools --channel=22/beta
+    ```
+3. Install `miriway` from the latest `beta` track:
+    ```
+    sudo snap refresh
+    sudo snap install miriway --classic --chanel=latest/beta
+    ```
```

Comment:
1. I think the testing of each snap deserves its own test plan, not trying to roll everything into one plan
2. Please don't forget Frame (confined-shell might also show things up)

Reply:
There are two things to test here:
1. Generic Mir functionality (in both a snap and non-snap context)
2. Specific functionality for Mir-based compositors

I am beginning to think that these are separate documents altogether. When you do a Mir release, you should validate the Generic Mir test plan. When you do an Ubuntu Frame release, you should validate the Ubuntu Frame test plan. And so on.

Reply:
Well, -ish. We recently had an issue where a Mir release broke Frame-specific functionality. These are downstreams, affected by Mir releases, so should be tested as well (not the other way around, of course).

Reply:
Yeah that's the tricky bit. Do we perform the entire Mir test plan for each Mir-based compositor? Or do we perform the entire Mir test plan on `miral-app`, and then only test compositor-specific features on each compositor?

I would vote for the former

Reply:
> I would vote for the former

I would say the former is impossible, you cannot everything here on every compositor: they may be limited in supported platforms, window management paradigms, enabling features (e.g. X11), ...

There is a core set of Mir functionality that can, and should, be tested with `miral-app`.

And each snap needs its own test plan running too.

Reply:
> Well, -ish. We recently had an issue where a Mir release broke Frame-specific functionality. These are downstreams, affected by Mir releases, so should be tested as well (not the other way around, of course).

Agreed, the Mir test plan should reference the tests needed for downstream projects.